heudiconv/parser.py

```diff
@@ -60,58 +62,87 @@ def find_files(regex, topdir=op.curdir, exclude=None,
             yield path
 
 
-def get_extracted_dicoms(fl):
-    """Given a list of files, possibly extract some from tarballs.
+def get_extracted_dicoms(
+        fl: typing.Iterable[str]
+        ) -> typing.ItemsView[typing.Optional[int], typing.List[str]]:
+    """Given a collection of files and/or directories, possibly extract 
+    some from tarballs.
 
-    For 'classical' heudiconv, if multiple tarballs are provided, they correspond
-    to different sessions, so here we would group into sessions and return
-    pairs  `sessionid`, `files`  with `sessionid` being None if no "sessions"
-    detected for that file or there was just a single tarball in the list
-    """
-    # TODO: bring check back?
-    # if any(not tarfile.is_tarfile(i) for i in fl):
-    #     raise ValueError("some but not all input files are tar files")
+    Parameters
+    ----------
+    fl : Iterable[str]
+        An iterable (e.g., list or tuple) of files to process.
+
+    Returns
+    -------
+    ItemsView[int | None, list[str]]
+        A tuple of keys (either integer or None) and list of strs representing
+          the absolute paths of files. 
 
-    # tarfiles already know what they contain, and often the filenames
-    # are unique, or at least in a unique subdir per session
-    # strategy: extract everything in a temp dir and assemble a list
-    # of all files in all tarballs
+    Notes
+    -----
+    For 'classical' heudiconv, if multiple archives are provided, they 
+    correspond to different sessions, so here we would group into sessions 
+    and return pairs `sessionid`, `files`  with `sessionid` being None if no 
+    "sessions" detected for that file or there was just a single tarball in the
+    list.
 
-    # cannot use TempDirs since will trigger cleanup with __del__
-    tmpdir = tempdirs('heudiconvDCM')
+    When contents of fl appear to be an unpackable archive, the contents are
+    extracted into utils.TempDirs(f'heudiconvDCM') and the mode of all 
+    extracted files is set to 700.
 
-    sessions = defaultdict(list)
-    session = 0
-    if not isinstance(fl, (list, tuple)):
+    When contents of fl are a list of files, they are treated as a single
+    session.
+    """
+    input_list_of_unpacked = any(fl) and all(
+        not t.endswith(_UNPACK_FORMATS) for t in fl
+        )
+    
+    if not (
+        input_list_of_unpacked or
+        all(t.endswith(_UNPACK_FORMATS) for t in fl)
+        ):
```

Comment:
```suggestion
    ):
```

<details>
<summary>well, we need to add some auto-blacking eventually etc, but ATM here and elsewhere let's stay consistent with dominant way -- aligning to original block indent, not nested:
</summary> 

```shell
❯ git grep -3  '^ *):$'
heudiconv/bids.py-            all([other[k] == v for k, v in self._entities.items()])
heudiconv/bids.py-            and self.extension == other.extension
heudiconv/bids.py-            and self.suffix == other.suffix
heudiconv/bids.py:        ):
heudiconv/bids.py-            return True
heudiconv/bids.py-        else:
heudiconv/bids.py-            return False
--
heudiconv/convert.py-    if (
heudiconv/convert.py-        op.exists(target_heuristic_filename) and
heudiconv/convert.py-        file_md5sum(target_heuristic_filename) != file_md5sum(heuristic.filename)
heudiconv/convert.py:    ):
heudiconv/convert.py-        # remake conversion table
heudiconv/convert.py-        reuse_conversion_table = False
heudiconv/convert.py-        lgr.info(
--
heudiconv/dicoms.py-    if mw.dcm_data[0x0008, 0x0016].repval in (
heudiconv/dicoms.py-        'Raw Data Storage',
heudiconv/dicoms.py-        'GrayscaleSoftcopyPresentationStateStorage'
heudiconv/dicoms.py:    ):
heudiconv/dicoms.py-        return
heudiconv/dicoms.py-    try:
heudiconv/dicoms.py-        file_studyUID = mw.dcm_data.StudyInstanceUID
--
heudiconv/external/dcmstack.py-    if not (
heudiconv/external/dcmstack.py-        ("No module" in e_str and "dicom" in e_str) or
heudiconv/external/dcmstack.py-        ('has been removed in pydicom version' in e_str)
heudiconv/external/dcmstack.py:    ):
heudiconv/external/dcmstack.py-        raise
heudiconv/external/dcmstack.py-    # a butt plug due to rename of dicom -> pydicom
heudiconv/external/dcmstack.py-    import sys
--
heudiconv/heuristics/bids_PhoenixReport.py-        if (
heudiconv/heuristics/bids_PhoenixReport.py-            'PhoenixZIPReport' in s.series_description
heudiconv/heuristics/bids_PhoenixReport.py-            and s.image_type[3] == 'CSA REPORT'
heudiconv/heuristics/bids_PhoenixReport.py:        ):
heudiconv/heuristics/bids_PhoenixReport.py-            info[phoenix_doc].append({'item': s.series_id})
heudiconv/heuristics/bids_PhoenixReport.py-        if 'scout' in s.series_description.lower():
heudiconv/heuristics/bids_PhoenixReport.py-            info[scout].append({'item': s.series_id})
--
heudiconv/tests/test_bids.py-        expected_prefix,
heudiconv/tests/test_bids.py-        simulation_function,
heudiconv/tests/test_bids.py-        match_param
heudiconv/tests/test_bids.py:):
heudiconv/tests/test_bids.py-    """
heudiconv/tests/test_bids.py-    Test find_compatible_fmaps_for_session.
heudiconv/tests/test_bids.py-
--
heudiconv/tests/test_bids.py-    expected_prefix,
heudiconv/tests/test_bids.py-    simulation_function,
heudiconv/tests/test_bids.py-    match_param
heudiconv/tests/test_bids.py:):
heudiconv/tests/test_bids.py-    """
heudiconv/tests/test_bids.py-    Test populate_intended_for.
heudiconv/tests/test_bids.py-    Parameters

```
</details>